### ISSUE TITLE
How to visualize Figure3

### ISSUE THREAD

**Issue by xiaobiaodu** [tianweiy/DMD2#2]
Hi

Thanks for your excellent work! 

Can I ask how you visualize Figure 3 in your paper? 

Thanks for your help in advance.

**Comment by tianweiy** [comment]
Thank you for the question! That one is not super automatic lol

<img width="1256" alt="Screenshot 2024-05-25 at 9 18 07 AM" src="https://github.com/tianweiy/DMD2/assets/38497390/b5ce2aaa-5fe0-4d38-a457-157a31f42eb2">

The bottom squirrel is a real image. I created the diffused image by alpha com-positioning a noise map and the original image. 

On the upper branch, I set the prompt to be "A punk rock squirrel in a studded leather jacket shouting into a microphone while standing on a boulder" on this [line](https://github.com/tianweiy/DMD2/blob/4269d3b3321cecf183125b37101555972373fda3/main/sd_image_dataset.py#L38).
We use an initial noise with random seed 0 (to be specific, I was doing torch.randn(2, 4, 128, 128, generator=torch.manual_seed(0))[1:], i.e. taking the second image for the random seed 0). 

The left-most image is the output by running one-step backward simulation. Basically, I set the selected_step = 1 and record the output at this [line](https://github.com/tianweiy/DMD2/blob/4269d3b3321cecf183125b37101555972373fda3/main/sd_unified_model.py#L133), after VAE decoding. 

The fake image as well as the corresponding real/fake denoised estimates are computed with a selected_step=2 (or 3 I don't exactly remember). We can dump these images at the following [line](https://github.com/tianweiy/DMD2/blob/4269d3b3321cecf183125b37101555972373fda3/main/sd_unified_model.py#L282)  

If you only interested in visualizing real and fake denoised estimates as well as their difference. This is already implemented in the wandb logging. 

Let me know if you have any further questions!
